src/artofillusion/polymesh/PolyMeshViewer.java

```diff
@@ -352,34 +352,19 @@ private void drawVertices()
 
         for (int i = 0; i < screenVert.length; i++)
         {
-            if (mirror)
-                ref = invVertTable[i];
-            else
-                ref = i;
+            if(!visible[i]) continue;
+            
+            Color color = selected[ mirror ? invVertTable[i] : i] ? selectedVertColor : vertColor;
```

Comment:
There is such a thing as being too concise. Nested ternaries are too concise. Please keep this as two separate lines.